src/components/CheckboxInput/components/Checkbox.tsx

```diff
@@ -0,0 +1,170 @@
+import React, {
+  FC, ChangeEvent, FocusEvent, ReactNode,
+} from 'react';
+import { Box } from '../../Box/Box';
+import { Icon } from '../../Icon/Icon';
+import { FontColor, FontSize } from '../../../types';
+
+export type CheckboxSize = 'sm' | 'md' | 'lg';
+
+export interface CheckboxProps {
+  /**
+   * The id attribute of the input.
+   */
+  id: string;
+  /**
+   * The checkbox input "checked" attribute.
+   */
+  isChecked: boolean;
+  /**
+   * Custom content to be displayed to right of checkbox.
+   */
+  label: string;
+  /**
+   * Callback function when input is changed.
+   */
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
+  /**
+   * Additional classes to add.
+   */
+  className?: string;
+  /**
+   * Mark the input field as invalid and display a validation message.
+   * Pass a string or node to render a validation message below the input.
+   */
+  error?: ReactNode;
+  /**
+   * If the input should be disabled and not focusable.
+   */
+  isDisabled?: boolean;
+  /**
+   * Determines if input is required or not. (Label will have an asterisk if required).
+   */
+  isRequired?: boolean;
+  /**
+   * Callback function when input is blurred.
+   */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /**
+   * Callback function when input is focused.
+   */
+  onFocus?: (event: FocusEvent<HTMLInputElement>) => void;
+  /**
+   * The size of the checkbox.
+   */
+  size?: CheckboxSize;
+  /**
+   * Additional props to be spread to checkbox input element
+   */
+  [x: string]: any; // eslint-disable-line
+}
+
+const SIZE_KEYS: { [key: string]: { iconSize: FontSize; height: string; }; } = {
+  sm: {
+    iconSize: 'lg',
+    height: '20px',
+  },
+  md: {
+    iconSize: 'xl',
+    height: '24px',
+  },
+  lg: {
+    iconSize: '2xl',
+    height: '36px',
+  },
+};
+
+export const Checkbox: FC<CheckboxProps> = ({
```

Comment:
all of our components should render with ref forwarding. So this should all be wrapped in a `React.forwardRef()`